pkg/crypto/xof.go

```diff
@@ -0,0 +1,37 @@
+package crypto
+
+import (
+	"errors"
+
+	"golang.org/x/crypto/sha3"
+)
+
+type PRFXOFHasher interface {
+	Hash(data []byte, outputLen int) ([]byte, error)
+}
+
+type cshakeHasher struct {
+	domain []byte
+	key    []byte
+}
+
+func NewCShakeHasher(key, domain []byte) PRFXOFHasher {
```

Comment:
suggest naming this variable "secret" or "secretKey" to disambiguate. Also, domain can be a string to prevent accidentally reversing the order of arguments here.